R/rj.b.R

```diff
@@ -167,6 +167,12 @@ RjClass <- R6::R6Class(
                 if (file.exists('/opt/local/bin/R'))
                     return(list(path='/opt/local/bin/R'))
 
+                path <- try(system2('which', args='R', stdout=TRUE), silent = TRUE)
+                if (!inherits(path, "try-error")) {
+                    if (!is.na(path) && file.exists(path)) {
```

Comment:
re: `is.na(path)`

are you aware of a situation where the path would come back as `NA`? i can't think of one (but if you know of one, great!).

one thing you need to handle is when R isn't found ... in that situation, `which` writes nothing to stdout (writing to stderr instead), and `system2()` will return a chracter string of zero length (`character(0)`). this is a problem, because `if (!is.na(path) ...` will poop itself trying to perform logic on an item of zero length.

when you're writing this, it's good to think through all the things that could go wrong, and make sure they're handled.

for example, maybe the `which` command doesn't work on someone's computer, or maybe `which R` fails ... to check that your code handles these situations, you can substitute in silly values, i.e.

```
system2('whichXXX', args='R', 
system2('which', args='RXXX', 
```

the `XXX` simulates situations where `which` or `R` can't be found.

cheers